backend/src/sync.js

```diff
@@ -91,6 +91,40 @@ async function saveBlocks(blocksInfo) {
                     });
                   })
                 ),
+                models.AccessKey.bulkCreate(
+                  blockInfo.transactions
+                    .filter(tx =>
+                      tx.actions.some(action => action.AddKey !== undefined)
+                    )
+                    .flatMap(tx => {
+                      const accountId = tx.receiver_id;
+                      return tx.actions
+                        .filter(action => action.AddKey !== undefined)
+                        .map(action => {
+                          let accessKeyType;
+                          if (
+                            action.AddKey.access_key.permission === "FullAccess"
+                          ) {
+                            accessKeyType = "FullAccess";
+                          } else if (
+                            action.AddKey.access_key.permission.hasOwnProperty(
+                              "FunctionCall"
+                            )
+                          ) {
+                            accessKeyType = "FunctionCall";
+                          } else {
+                            throw new Error(
+                              "permission type is either FullAccess nor FunctionCall"
```

Comment:
I would really have a hard time understanding this message if it would appear in the logs, wouldn't you?

Reply:
what is your suggestion? @frol

Reply:
Well, I would just drop the manual parsing altogether and just check that if that is a string, that is the accessKeyType, otherwise, use the `Object.keys()[0]` as the accessKeyType (ensuring that it is not undefined).

Error messages should be helpful, and to be so, we should include the relevant information into them, e.g.:

> Unexpected error during access key permission parsing in transaction ${tx.hash}: the permission type is expected to be a string or an object with a single key, but '${JSON.stringify(permission)}' found.